content/vendor/guides/persistent-datastores.md

```diff
@@ -0,0 +1,940 @@
+---
+date: "2020-02-20T00:00:00Z"
+lastmod: "2020-02-20T00:00:00Z"
+title: "Integrating Persistent Datastores"
+weight: "1006"
+---
+
+In this guide, we'll review best practices for integrating persistent stores like databases, queues, and caches. We'll explore ways to give your end user the option to either embed an instance with the application, or connect your appplication to an external instance that they will manage. We'll use a PostgreSQL database as an example, configuring an example app to connect. The latter part of this guide starts to explore advanced topics like custom preflight checks, workload coordination, credential management, and refactoring your application's user-facing configuration.
+
+It is split into 5 sections:
+
+- [The Example Application](#the-example-application)
+- [User-Facing Configuration](#user-facing-configuration)
+- [Embedding a Database](#embedding-a-database)
+- [Connecting to an External Database](#connecting-to-an-external-database)
+
+### Prerequisites
+
+You should have completed one of the [Getting Started Guides](../#getting-started), this guide assumes you have a running instance of `kotsadm` to iterate against in either an existing or embedded cluster, and a local git checkout of your KOTS app manifests.
+
+### Accompanying Code Examples
+
+A full example of the code for this guide can be found in the [kotsapps repository](https://github.com/replicatedhq/kotsapps/tree/master/persistent-datastores).
+
+* * *
+
+## The Example Application
+
+For demonstration purposes, we'll use a simple app that connects to a postgres database via the `psql` CLI. Once you've finished this guide, you should feel confident replacing it with any Kubernetes workload(s) that need to connect to a database. The deployment we'll use can be seen below:
+
+```yaml
+# pg-consumer.yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: pg-consumer 
+spec:
+  selector:
+    matchLabels:
+      app: pg-consumer
+  template:
+    metadata:
+      labels: 
+        app: pg-consumer
+    spec:
+      containers:
+        - name: pg-consumer
+          image: postgres:10
+          # connect to the database every 20 seconds
+          command:
+            - /bin/sh
+            - -ec
+            - |
+              while :; do 
+                 sleep 20
+                 PGPASSWORD=${DB_PASSWORD} \
+                 psql --host ${DB_HOST} \
+                      --port ${DB_PORT} \ 
+                      --user ${DB_USER} \
+                      --dbname ${DB_NAME} \
+                      --command 'SELECT NOW()'
+              done
+          # hard coded for now, we'll wire these up later
+          env:
+            - name: DB_HOST
+              value: postgres
+            - name: DB_PORT
+              value: 5432
```

Comment:
This port needs to be quoted.

Reply:
Actually, it looks like all these values need to be in quotes (e.g., 'postgres').